File: rust/a2021/src/bin/01.rs
fn main() {
    println!(
        "part 1: {}",
        common::get_problem_lines(2021, 1)
            .map(|l| l.parse::<i32>().unwrap())
            .collect::<Vec<i32>>()
            .windows(2)
            .filter(|s| s[0] < s[1])
            .count()
    );
    println!(
        "part 2: {}",
        common::get_problem_lines(2021, 1)
            .map(|l| l.parse::<i32>().unwrap())
            .collect::<Vec<i32>>()
            .windows(4)
            .filter(|s| s[0] < s[3])
            .count()
    );
}

File: python/a2019/01.py
from common import gpl

f = lambda i: (i // 3) - 2

p1 = 0
p2 = 0
for l in gpl():
    p1 += f(int(l))
    d = f(int(l))
    while d > 0:
        p2 += d
        d = f(d)

print(f"part 1: {p1}")
print(f"part 2: {p2}")
